examples/pty/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# PTY (pseudo-terminals) are available on most UNIX-like
+# systems but for simplicity we only check macOS and Linux.
+# Windows has an entirely different API for terminals which this
+# demo does not support yet.
+set(SUPPORTED_SYSTEMS Darwin Linux)
+if(NOT CMAKE_SYSTEM_NAME IN_LIST SUPPORTED_SYSTEMS)
+    message(FATAL_ERROR "The PTY example supports macOS and Linux only.")
```

Comment:
Maybe it should be a warning instead, because this directory is added by the top-level CMakeLists.txt and we shouldn't prevent Windows users to building the library itself.

Reply:
That's probably sensible. A related issue I have is that BUILD_EXAMPLES is normally ON, which means we always build examples with `make all`. I think the user should have to specify if they want to build the examples rather than building them by default.

Reply:
I thought we wanted to avoid `EXCLUDE_FROM_ALL`, so the examples are built by default when building the current project as the main project (#26). At least I'll remove the fatal error.

Reply:
OK, fair enough. We can just remove the error for now.

However, I don't really understand why we wanted to avoid `EXCLUDE_FROM_ALL` in this case, as `EXCLUDE_FROM_ALL` is basically intended for optional examples. That use case is even mentioned in the [cmake docs](https://cmake.org/cmake/help/latest/command/add_subdirectory.html). I know we want to change the way we use `EXCLUDE_FROM_ALL` in Mbed OS, because we are forced to abuse it due to the way the Mbed OS source tree is structured. Anyway, we can discuss that more on Monday, as it's a design discussion which is out of scope for this PR.